wasm/include/rtc/configuration.hpp

```diff
@@ -25,8 +25,27 @@
 
 namespace rtc {
 
+struct IceServer {
+	enum class Type { Stun, Turn };
+	
+	// STUN
+	IceServer(string hostname_, uint16_t port_)
+		: hostname(std::move(hostname_)), port(port_), type(Type::Stun) {}
+	
+	// TURN
+	IceServer(string hostname_, uint16_t port_, string username_, string password_)
+		: hostname(std::move(hostname_)), port(port_), type(Type::Turn), username(std::move(username_)),
+      	  password(std::move(password_)) {}
```

Comment:
My opinion is, in case you think the parsing logic won't change for a very long time, adding a parsing constructor to datachannel-wasm would be useful. But if not, it may unnecessarily cause a compatibility issue later, considering that the url parser can be implemented in one's application layer.

Reply:
I don't think the parsing will ever change, at the maximum there will bug bug fixes to report here. I think I'll add it later.

Reply:
That would be great!!